ios/RCTOneSignal/RCTOneSignal.m

```diff
@@ -78,7 +78,23 @@ - (void)sendEvent:(NSString *)eventName withBody:(NSDictionary *)body {
 }
 
 - (void)onPushSubscriptionDidChangeWithState:(OSPushSubscriptionChangedState * _Nonnull)state {
-    [self sendEvent:OSEventString(SubscriptionChanged) withBody:[state jsonRepresentation]];
+    NSMutableDictionary *result = [NSMutableDictionary new];
+    
+    //Previous state
+    NSMutableDictionary *previousObject = [NSMutableDictionary new];
+    previousObject[@"token"] = (state.previous.token && ![state.previous.token isEqualToString:@""]) ? state.previous.token : [NSNull null];
```

Comment:
These lines get pretty complex, we can consider refactoring these into getter helpers, that can also be used for the User State observer., once this PR and that one are combined / merged.

Reply:
We should also update the PR description to callout the API "breaking-sorta" change that the push subscription observer will be passed nullable properties now.

Reply:
Thanks for flagging these! I've edited the PR description with the minor breaking change and will be sure to include it in release notes. Refactoring mentioned above will happen in a future PR.